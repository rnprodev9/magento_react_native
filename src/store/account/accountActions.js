import {
  MAGENTO,
  LOGIN_SUCCESS,
  UPDATE_CUSTOMER,
  ACTION_USER_LOGOUT,
} from '../../constants';

/**
 * Dispatch an action, containing logged in user token
 * token which is used as session token in app
 *
 * @param {string} token - Logged in user token
 */
export const loginSuccess = token => ({
  type: LOGIN_SUCCESS,
  payload: {
    token,
  },
});

/**
 * Dispatch an action to fetch logged in user details
 */
export const getCurrentCustomer = () => ({
  type: MAGENTO.CURRENT_USER_REQUEST,
});

/**
 * Update the user details in reducer
 */
export const updateCustomer = customer => ({
  type: UPDATE_CUSTOMER,
  payload: {
    customer,
  },
});

/**
 * Dispatch an action, to logout user and clear caches
 * and session token.
 */
export const logout = () => ({
  type: ACTION_USER_LOGOUT,
});

/**
 * Fetch logged in user orders
 *
 * @param {number} customerId  - Logged in user id
 * @param {number} offset      - Number of items already fetched
 */
export const getOrders = (customerId, offset = 0) => ({
  type: MAGENTO.GET_ORDERS_REQUEST,
  payload: { customerId, offset },
});
