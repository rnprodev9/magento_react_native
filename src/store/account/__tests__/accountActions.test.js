import { loginSuccess } from '../accountActions';

describe('accountActions', () => {
  test('loginSuccess()', () => {
    // Setup
    const token = 'abc123@#$';

    // Exercise
    const result = loginSuccess(token);

    // Verify
    expect(result).toMatchSnapshot();
  });
});
