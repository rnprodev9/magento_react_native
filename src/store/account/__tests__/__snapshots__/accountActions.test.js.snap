// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accountActions loginSuccess() 1`] = `
Object {
  "payload": Object {
    "token": "abc123@#$",
  },
  "type": "LOGIN_SUCCESS",
}
`;
